src/nvim/api/vim.c

```diff
@@ -987,6 +987,78 @@ Buffer nvim_create_buf(Boolean listed, Boolean scratch, Error *err)
   return buf->b_fnum;
 }
 
+/// Open a new window.
+///
+/// Currently this is used to open floating and external windows.
+/// Floats are windows that are drawn above the split layout, at some anchor
+/// position in some other window. Floats can be draw internally or by external
+/// GUI with the |ui-multigrid| extension. External windows are only supported
+/// with multigrid GUIs, and are displayed as separate top-level windows.
+///
+/// Exactly one of `external` and `relative` must be specified.
+///
+/// @param buffer handle of buffer to be displayed in the window
+/// @param enter whether the window should be entered (made the current window)
+/// @param width width of window (in character cells)
+/// @param height height of window (in character cells)
+/// @param options dict of options for configuring window positioning
+///                accepts the following keys:
+///     `external` GUI should display the float as an external
+///         top-level window. Currently accepts no other positioning options
+///         together with this.
+///     `relative`: If set, the window becomes floating window. The window
+///         will be placed with row,col coordinates relative one of the
+///         following:
+///        "editor" the global editor grid (default)
+///        "window" window id for placing relative another window.
+///        "cursor" the cursor position in current window.
+///     `anchor`:   the corner of the float that the row,col position defines
+///        "NW" north-west (default)
+///        "NE" north-east
+///        "SW" south-west
+///        "SE" south-east
+///     `unfocusable`: The window will not be focused by wincmds and
+///         mouse events. It can still be made current by API calls.
```

Comment:
Negative logic is confusing, so `focusable` is a better name. It's ok if the default is `true`.

Reply:
"made current" means "made the current window", "entered" would be established verb for this. "focused" means entered due to user input being directed towards the window, i e by clicking inside the window.